### DISCUSSION TITLE
Generate PDF from html - only getting screenshot, not whole report

### DISCUSSION THREAD

**Discussion by memetzgz** [datapane/datapane#102] [Help]
Hello all, I just started experimenting with Datapane for static report generation, and I am really liking it so far.  This has been an obvious (and ignored) need within the python ecosystem, so kudos to the developers for taking on the task!

I would like to be able to share my reports in PDF format, rather than HTML.  I've tried printing to PDF from the HTML document in both Chrome and Safari, and what I get is a single page screenshot, basically, instead of the whole html page being printed.  

Is there some way to get the full PDF out of the HTML document, either via printing or exporting to PDF, or some other mechanism?  

Thanks for your help!

**Comment by lanthias** [comment]
Thanks for this, and for the kind words! 

So we can assess, it would be really helpful to know the situations where PDF would be a better fit than HTML - of course the PDF would lose the interactive elements and the HTML is largely standalone as a PDF is, so I would love to know the scenarios where it would be more suitable so we can work around the use-case.

Thanks!
Leo

**Reply by memetzgz** [reply]
Hi Leo,
Thanks for your reply! I'd be happy to provide additional detail if needed, but I can address the use cases I have as Director of Research and Analytics at a small-ish startup.  

**The use cases are the following:**
1.  sharing analyses internally with non-technically oriented folks.  They don't need the interactivity of an html report.  They want a paginated PDF that emulates a hard copy.  In fact, they might even print it out to have a hard copy.  The paper-based, hard copy report is the paradigm for people in our organization who are not developers (think sales, marketing, client services, admin).  For cases where we need interactivity, I'd rather go full-on dashboard, with selectors, etc.
2. Similarly, generating and sharing reports for clients.  Here, too, the paradigm is old-school hard copy.  These reports need to look a little nicer than internal ones, but the constraints are similar - paginated; 8.5 x 11 portrait, or possibly a landscape orientation in a typical PowerPoint slide size; could be printed out and still look reasonably good.  We're not going to be able to simply share HTML files with them, that's considered too insecure and it's just too "weird" for our clients (they are not in the tech space).

**Here's some context about why it's been problematic to address those use cases:** 
I use a Python stack.  My company's front-end developers need to work on our product, I'm never going to get their time to build out reports.  

We are in a healthcare-related field, so data and system security is #1, and we have a lot of constraints we have to work around accordingly.  We can't host stuff in the cloud unless it's in our our space. 

I want to avoid having to copy and paste tables and graphs into PowerPoint or Word, just to be able to share them in a format that is secure and that people want to get stuff in.  I want to be able to automate as much as possible, right down to the last mile.  People tell me I could use RMarkdown; maybe that's true, but I would really prefer not to have to add R to the mix, if I can do it in Python.

Theoretically I should be able to export a Jupyter Notebook to HTML or to a PDF, and just hide all the input cells.  In reality despite much effort I've not gotten this to work well, and the result is not paginated, so graphs get spread over page breaks, etc.

That was long, but I hope it gives you a sense of things from the end user point of view that is helpful!  I don't think I am an outlier here, I feel like this is a common business use case, especially for organizations trying to move away from Excel-based analytics.  

Many thanks for your consideration of these use cases!
Maureen

**Reply by johnmicahreid** [reply]
Hi Maureen, John here (a Product Manager at Datapane). Thanks for the detailed write-up, it's really helpful! I'd like to dig into your use case a little further: 
- What kinds of charts are contained in your reports, and are they mostly static? One of our concerns with PDF export is data loss i.e. if you have a complex interactive chart which contains individual data points that can be viewed by hovering over them, that experience won't convert well into a PDF
- Does your team need to save all reports as PDFs in some sort of archive, or is it more that certain people just prefer the PDF/printed format?
- Would you be willing to try an on-premise version of Datapane? Our Community and Teams products allow you to upload a Datapane report to a server (which you could host in your own cloud) and share it with anyone via an invite or a secure token, which we've found is quite convenient for non-technical folks to view without any setup or installation.

**Comment by isaipina** [comment]
Hi @memetzgz 

The best workaround is to use Microsoft Edge and use ctrl + p when you have the .HTML exported by Datapane. 
In the Print Dialog box, you can select the configuration that fit with your report size.  It's imperative select the "Background graphics" option.
